src/apps/silo-finance/common/silo-finance.silo.contract-position-fetcher.ts

```diff
@@ -0,0 +1,233 @@
+  ): Promise<SiloContractPositionDataProps> {
+    const siloLens = this.getSiloLens();
+    const definition = params.definition;
+
+    const tokenContractsFlattened = definition.tokensFlattened.map(token =>
+      this.siloFinanceContractFactory.erc20({ address: token, network: this.network }),
+    );
+    const totalShares = await Promise.all(
+      tokenContractsFlattened.map(tokenContract => params.multicall.wrap(tokenContract).totalSupply()),
+    );
+    const totalUnderlyings = await Promise.all(
+      definition.tokens.map(async (_, idx) => {
+        const asset = definition.assets[idx];
+        const assetStorage = definition.assetStorages[idx];
+        const totalDepositWithInterest = await siloLens.totalDepositsWithInterest(params.address, asset);
+        const totalBorrowsWithInterest = await siloLens.totalBorrowAmountWithInterest(params.address, asset);
+        return [totalDepositWithInterest, assetStorage.collateralOnlyDeposits, totalBorrowsWithInterest];
+      }),
+    );
+    const supplyApys = await Promise.all(
+      definition.assets.map(asset => params.multicall.wrap(siloLens).depositAPY(params.address, asset)),
+    );
+    const borrowApys = await Promise.all(
+      definition.assets.map(asset => params.multicall.wrap(siloLens).borrowAPY(params.address, asset)),
+    );
+
+    return {
+      totalUnderlyings: totalUnderlyings.flat(),
+      totalShares: totalShares,
+      supplyApys: supplyApys,
+      borrowApys: borrowApys,
+      tokensFlattened: definition.tokensFlattened,
+    };
+  }
+
+  async getStatsItems({
+    contractPosition: { dataProps, tokens },
+    definition,
+  }: GetDisplayPropsParams<Silo, SiloContractPositionDataProps, SiloContractPositionDefinition>): Promise<
+    StatsItem[] | undefined
+  > {
+    return definition.assets
+      .map((_, idx) => {
+        const tokenLabel = getLabelFromToken(tokens[idx * 3]);
+        return [
+          {
+            label: `${tokenLabel} Supply APY`,
+            value: buildPercentageDisplayItem(+formatEther(dataProps.supplyApys[idx]) * 100),
+          },
+          {
+            label: `${tokenLabel} Borrow APY`,
+            value: buildPercentageDisplayItem(+formatEther(dataProps.borrowApys[idx]) * 100),
+          },
+        ];
+      })
+      .flat();
+  }
+
+  async getTokenDefinitions(
+    params: GetTokenDefinitionsParams<Silo, SiloContractPositionDefinition>,
+  ): Promise<UnderlyingTokenDefinition[] | null> {
+    return params.definition.assets
+      .map(asset => {
+        return [
+          {
+            address: asset,
+            network: this.network,
+            metaType: MetaType.SUPPLIED, // sTOKEN
+          },
+          {
+            address: asset,
+            network: this.network,
+            metaType: MetaType.SUPPLIED, // spTOKEN
+          },
+          {
+            address: asset,
+            network: this.network,
+            metaType: MetaType.BORROWED, // dTOKEN
```

Comment:
You're defining the underlying tokens, but then getting balances from another set of tokens ?